contracts/proposals/ListOnZoraProposal.sol

```diff
@@ -167,40 +152,33 @@ contract ListOnZoraProposal is ZoraHelpers {
 
     // Either cancel or finalize a Zora auction.
     function _settleZoraAuction(
-        uint256 auctionId,
         uint40 minExpiry,
-        IERC721 token,
+        address token,
         uint256 tokenId
     ) internal override returns (ZoraAuctionStatus statusCode) {
-        // Getting the state of an auction is super expensive so it seems
-        // cheaper to just let `endAuction()` fail and react to the error.
-        try ZORA.endAuction(auctionId) {
-            // Check whether auction cancelled due to a failed transfer during
-            // settlement by seeing if we now possess the NFT.
-            if (token.safeOwnerOf(tokenId) == address(this)) {
-                emit ZoraAuctionFailed(auctionId);
-                return ZoraAuctionStatus.Cancelled;
+        IReserveAuctionCoreEth.Auction memory auction = ZORA.auctionForNFT(token, tokenId);
+        if (auction.seller != address(this)) {
+            // Auction has already been settled
+            emit ZoraAuctionSold(token, tokenId);
+            return ZoraAuctionStatus.Sold;
+        }
+        if (auction.firstBidTime == 0) {
+            if (minExpiry > block.timestamp) {
+                revert ZoraListingNotExpired(token, tokenId, minExpiry);
             }
-        } catch (bytes memory errData) {
-            bytes32 errHash = keccak256(errData);
-            if (errHash == AUCTION_HASNT_BEGUN_ERROR_HASH) {
-                // No bids placed.
-                // Cancel if we're past the timeout.
-                if (minExpiry > uint40(block.timestamp)) {
-                    revert ZoraListingNotExpired(auctionId, minExpiry);
-                }
-                ZORA.cancelAuction(auctionId);
-                emit ZoraAuctionExpired(auctionId, minExpiry);
-                return ZoraAuctionStatus.Expired;
-            } else if (errHash != AUCTION_DOESNT_EXIST_ERROR_HASH) {
-                // Otherwise, we should get an auction doesn't exist error,
-                // because someone else must have called `endAuction()`.
-                // If we didn't then something is wrong, so revert.
-                errData.rawRevert();
+            // minExpiry passed with no bids
+            ZORA.cancelAuction(token, tokenId);
+            emit ZoraAuctionExpired(token, tokenId, minExpiry);
+            return ZoraAuctionStatus.Expired;
+        } else {
+            if (block.timestamp >= auction.duration + auction.firstBidTime) {
```

Comment:
Nit, but since we potentially use `auction.duration + auction.firstBidTime` twice we can cache it